my_compassion/controllers/my2_my_donations.py

```diff
@@ -0,0 +1,136 @@
+import math
+from collections import defaultdict
+from datetime import datetime, timedelta
+
+from odoo import fields
+from odoo.http import request, route
+
+from odoo.addons.portal.controllers.portal import CustomerPortal
+
+
+class MyDonationController(CustomerPortal):
+    def get_paid_invoices_filter(self, partner):
+        paid_invoices_filter = [
+            ("partner_id", "=", partner.id),
+            ("payment_state", "=", "paid"),
+            ("move_type", "=", "out_invoice"),
+            ("amount_total", "!=", 0),
+        ]
+        return paid_invoices_filter
+
+    def get_paid_invoices_subset(self, partner, offset, amount):
+        paid_invoices_subset = (
+            request.env["account.move"]
+            .sudo()
+            .search(
+                self.get_paid_invoices_filter(partner),
+                offset=offset,
+                limit=amount,
+            )
+        )
+        return paid_invoices_subset
+
+    def get_paid_invoices_amount(self, partner):
+        number_of_paid_invoices = (
+            request.env["account.move"]
+            .sudo()
+            .search_count(self.get_paid_invoices_filter(partner))
+        )
+        return number_of_paid_invoices
```

Comment:
Typically, these would be private class methods (this is more stylistic than anything, but communicates intent clearly).

```suggestion
    @classmethod
    def _get_paid_invoices_filter(cls, partner):
        paid_invoices_filter = [
            ("partner_id", "=", partner.id),
            ("payment_state", "=", "paid"),
            ("move_type", "=", "out_invoice"),
            ("amount_total", "!=", 0),
        ]
        return paid_invoices_filter

    @classmethod
    def _get_paid_invoices_subset(cls, partner, offset, amount):
        paid_invoices_subset = (
            request.env["account.move"]
            .sudo()
            .search(
                cls.get_paid_invoices_filter(partner),
                offset=offset,
                limit=amount,
            )
        )
        return paid_invoices_subset

    @classmethod
    def _get_paid_invoices_amount(cls, partner):
        number_of_paid_invoices = (
            request.env["account.move"]
            .sudo()
            .search_count(cls.get_paid_invoices_filter(partner))
        )
        return number_of_paid_invoices
```